### ISSUE TITLE
ANN401 does not trigger on concatenated string literals

### ISSUE THREAD

**Issue by robsdedude** [astral-sh/ruff#19184]
### Summary

```python
from typing import Union, Any

def f1(a: "Union[str, bytes, Any]") -> None: ...  # causes ANN401
def f2(a: "Union[str, bytes, A" "ny]") -> None: ...  # no lint
```

https://play.ruff.rs/e35cf953-b4be-467b-96d9-c4639fe736f9

Arguably, f2 should also be flagged with `ANN401`.

### Version

v0.12.2

**Comment by MichaReiser** [comment]
I don't think this is worth supporting given that string annotations are uncommon and concatenated string annotations are even less common.
